helpers/constants.js

```diff
@@ -110,7 +110,10 @@ var constants = {
 		offset: 1451520, // Start rewards at block (n)
 		distance: 3000000, // Distance between each milestone
 	},
-	// WARNING: When changing totalAmount you also need to change getBlockRewards(int) SQL function!
+	// Every version starts on provided height
+	blockVersions: [
```

Comment:
In `exceptions.js` there is a special `precedent` field design to deal with legacy issues like block versioning https://github.com/LiskHQ/lisk/blob/development/config/default/exceptions.js#L51.
Please move the constant to exceptions.

Reply:
1. That structure is not present in `rc.2` version.
2. I don't agree that block versions should be moved to exceptions. While the thing we already have there (disabling particular transactions) is a precedent (happened once and will not happen again), block version is not. That block version will be used in every future hard-fork and will become integral part of the application. It's why I think it should stay in constants, like block rewards milestones for example.

Reply:
The feature `blockVersion` is similar to `rewards.milestones` where we specify some values for milestones on particular block height. 

```
	rewards: {
		milestones: [
			500000000, // Initial Reward
			400000000, // Milestone 1
			300000000, // Milestone 2
			200000000, // Milestone 3
			100000000, // Milestone 4
		],
```
Similar is for `blockVersions`
```
blockVersions: [
     1, // Initial block version to 0
      1334445, Block Version 1
]
```

So this options seems to be perfect to reside in constants.

Reply:
https://github.com/LiskHQ/lisk/pull/2233#discussion_r202700076

In constants, we store the Lisk network parameters. The newly created network wouldn't contain any hard forks, nor exceptions. The same behaviour. The network is parametrized by 'constants' which supposed to be... constant, not changed by _"become integral part of the application"_.

https://github.com/LiskHQ/lisk/pull/2233#discussion_r202720478
Changing rewards in milestones is the network feature that decreases the inflation over time. 

Keeping on track of all of the block versions (hard forks) is the not the network parameter, nor the feature. 
It is a legacy change needed to be introduced to allow the network to continue. Same as all of the other exceptions. I insist to move it to exceptions under the newly created "precedent" key, that will be merged with dapp in/out disable while merging with 1.1.0 version.